packages/flutter/lib/src/painting/borders.dart

```diff
@@ -151,20 +159,29 @@ class BorderSide {
   /// A hairline black border that is not rendered.
   static const BorderSide none = BorderSide(width: 0.0, style: BorderStyle.none);
 
-  /// The direction of where the border will be drawn relative to the container.
-  final StrokeAlign strokeAlign;
+  /// The relative position of the stroke on a [BorderSide] in a [Border] or [OutlinedBorder].
+  /// When set to [StrokeAlign.inside], the stroke is drawn completely inside the widget.
+  /// For [StrokeAlign.center] and [StrokeAlign.outside], a property such as
+  /// [Container.clipBehavior] can be used in an outside widget to clip it.
+  /// If [Container.decoration] has a border, the container may incorporate
+  /// [width] as additional padding:
+  /// - [StrokeAlign.inside] provides padding with full [width].
```

Comment:
I think first time readers of this will be confused by the reference to `StrokeAlign`, since it's not referenced in any other way. Their first thought on seeing that would probably be "Oh? What does the `StrokeAlign` class do, maybe I need that?"

I still prefer the regular doubles, and referring to `[strokeAlignInside]` instead, since it's a value on this same class.

Reply:
I agree from the logical point on the `strokeAlignInside`. But development experience got so much worse for me. Typing `c`+enter is better when it doesn't add `canLerp`. shift + alt + back I only had to rewrite what I needed, now need a lot more.

I think this comment appeared when we decided to use doubles. It just occurred to me it is a copy of what is inside StrokeAlign. I'll try to reword a bit, let's see if we can make it work. If not, you know, we can go with strokeAlignInside, but I really would rather not.

I would be willing to

Reply:
Changed to:
<img width="655" alt="image" src="https://user-images.githubusercontent.com/351125/182205327-b9ef6890-3d1b-49f8-8735-657da05d7ac4.png">